packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1137,127 +1137,156 @@ List<PluginInterfaceResolution> resolvePlatformImplementation(
-                'To fix this issue, add one of these dependencies to pubspec.yaml.\n'
-        );
         hasResolutionError = true;
-        continue;
       }
     }
 
-    finalResolution.addAll(
-      pluginResolution.map((Plugin plugin) =>
+    pluginResolutions.addAll(
+      pluginResolution.values.map((Plugin plugin) =>
           PluginInterfaceResolution(plugin: plugin, platform: platformKey)),
     );
   }
   if (hasResolutionError) {
     throwToolExit('Please resolve the plugin implementation selection errors');
   }
-  return finalResolution;
+  return pluginResolutions;
+}
+
+(String? resolutionPluginName, String? defaultImplementation) _getPluginImplementationCandidate(Plugin plugin, String platformKey) {
+  String? defaultImplementation = plugin.defaultPackagePlatforms[platformKey];
+  if (plugin.platforms[platformKey] == null && defaultImplementation == null) {
+    // The plugin doesn't implement this platform.
+    return (null, null);
+  }
+  String? implementsPackage = plugin.implementsPackage;
+  final bool hasInlineDartImplementation =
+      plugin.pluginDartClassPlatforms[platformKey] != null &&
+          plugin.pluginDartClassPlatforms[platformKey] != 'none';
+  if (implementsPackage == null || implementsPackage.isEmpty) {
+    if (defaultImplementation == null && !hasInlineDartImplementation) {
+      // Skip native inline PluginPlatform implementation
+      return (null, null);
+    }
+    if (defaultImplementation != null) {
+      return (null, defaultImplementation);
+    } else {
+      // An app-facing package (i.e., one with no 'implements') with an
+      // inline implementation should be its own default implementation.
+      // Desktop platforms originally did not work that way, and enabling
+      // it unconditionally would break existing published plugins, so
+      // only treat it as such if either:
+      // - the platform is not desktop, or
+      // - the plugin requires at least Flutter 2.11 (when this opt-in logic
+      //   was added), so that existing plugins continue to work.
+      // See https://github.com/flutter/flutter/issues/87862 for details.
+      final bool isDesktop = platformKey == 'linux' || platformKey == 'macos' || platformKey == 'windows';
+      final semver.VersionConstraint? flutterConstraint = plugin.flutterConstraint;
+      final semver.Version? minFlutterVersion = flutterConstraint != null &&
+          flutterConstraint is semver.VersionRange ? flutterConstraint.min : null;
+      final bool hasMinVersionForImplementsRequirement = minFlutterVersion != null &&
+          minFlutterVersion.compareTo(semver.Version(2, 11, 0)) >= 0;
+      if (!isDesktop || hasMinVersionForImplementsRequirement) {
+        // Given: implementsPackage == null, defaultImplementation == null, hasInlineDartImplementation == true
+        implementsPackage = plugin.name;
+        defaultImplementation = plugin.name;
+        return (implementsPackage, defaultImplementation);
```

Comment:
As `hasInlineDartImplementation` has now a unified meaning, there's no case, where these variable can alter, so can return immediately.

Reply:
Here and below: if you think the logic is not sufficiently self-explanatory that it needs a comment to explain why it's doing what it's doing, that needs to be a code comment, where future readers will be able to see it, rather than a PR comment than only a handful of people will ever read.

Reply:
I'm aware of that. I just wrote it down to explain the validity of the refactor, not the current state. The reader doesn't know about what was written there before. I'm not trying to explain the logic of the resolution with this comment. The same applies for the comment below.